src/matchers.rs

```diff
@@ -363,38 +363,44 @@ where
     V: TryInto<HeaderValue>,
     <V as TryInto<HeaderValue>>::Error: std::fmt::Debug,
 {
-    let values = values
-        .into_iter()
-        .filter_map(|v| v.try_into().ok())
-        .collect::<HeaderValues>();
     HeaderExactMatcher::new(key, values)
 }
 
 impl HeaderExactMatcher {
-    pub fn new<K, V>(key: K, value: V) -> Self
+    pub fn new<K, V>(key: K, values: Vec<V>) -> Self
     where
         K: TryInto<HeaderName>,
         <K as TryInto<HeaderName>>::Error: std::fmt::Debug,
-        V: TryInto<HeaderValues>,
-        <V as TryInto<HeaderValues>>::Error: std::fmt::Debug,
+        V: TryInto<HeaderValue>,
+        <V as TryInto<HeaderValue>>::Error: std::fmt::Debug,
     {
         let key = key.try_into().expect("Failed to convert to header name.");
-        let value = value
-            .try_into()
-            .expect("Failed to convert to header value.");
-        Self(key, value)
+        let values = values
+            .into_iter()
+            .map(|value| {
+                value
+                    .try_into()
+                    .expect("Failed to convert to header value.")
+            })
+            .collect();
+        Self(key, values)
     }
 }
 
 impl Match for HeaderExactMatcher {
     fn matches(&self, request: &Request) -> bool {
-        match request.headers.get(&self.0) {
-            None => false,
-            Some(values) => {
-                let headers: Vec<&str> = self.1.iter().map(HeaderValue::as_str).collect();
-                values.eq(headers.as_slice())
-            }
+        let values = request
+            .headers
+            .get_all(&self.0)
+            .iter()
+            .filter_map(|v| v.to_str().ok())
+            .flat_map(|v| v.split(','))
+            .map(|v| HeaderValue::from_str(v.trim()).unwrap())
```

Comment:
We can't unwrap here.

Reply:
It's fine to unwrap in matcher creation, since that would bubble up into the test setup code. But we can't unwrap in the matching itself, otherwise that's going to be pretty confusing to debug.